docs/modules/ROOT/pages/documentation/user-guides/custom-notifier.adoc

```diff
@@ -0,0 +1,570 @@
+                <_include>-osgi.bundle</_include>
+            </instructions>
+            <excludeDependencies>tools-jar</excludeDependencies>
+            <supportedProjectTypes>
+                <supportedProjectType>glassfish-jar</supportedProjectType>
+                <supportedProjectType>jar</supportedProjectType>
+            </supportedProjectTypes>
+        </configuration>
+    </plugin>
+    <plugin>
+        <groupId>org.glassfish.build</groupId>
+        <artifactId>command-security-maven-plugin</artifactId>
+        <version>1.0.10.payara-p1</version>
+        <configuration>
+            <isFailureFatal>false</isFailureFatal>
+        </configuration>
+    </plugin>
+    <plugin>
+        <groupId>org.glassfish.hk2</groupId>
+        <artifactId>config-generator</artifactId>
+        <version>2.5.0-b53</version>
+        <executions>
+            <execution>
+                <goals>
+                    <goal>generate-injectors</goal>
+                </goals>
+            </execution>
+        </executions>
+        <configuration>
+            <supportedProjectTypes>jar,glassfish-jar</supportedProjectTypes>
+        </configuration>
+    </plugin>
+    <plugin>
+        <groupId>org.glassfish.hk2</groupId>
+        <artifactId>osgiversion-maven-plugin</artifactId>
+        <version>2.6.1.payara-p1</version>
+        <executions>
+            <execution>
+                <id>default-compute-osgi-version</id>
+                <phase>process-classes</phase>
+                <goals>
+                    <goal>compute-osgi-version</goal>
+                </goals>
+            </execution>
+        </executions>
+        <configuration>
+            <dropVersionComponent>qualifier</dropVersionComponent>
+            <versionPropertyName>project.osgi.version</versionPropertyName>
+        </configuration>
+    </plugin>
+</plugins>
+----
+
+The Payara specific versions of these plugins can be found in our Patched repository on Github.
+
+----
+  <pluginRepositories>
+        <pluginRepository>
+            <releases>
+                <enabled>true</enabled>
+            </releases>
+            <snapshots>
+                <enabled>false</enabled>
+            </snapshots>
+            <id>payara-patched-externals</id>
+            <name>Payara Patched Externals</name>
+            <url>https://raw.github.com/payara/Payara_PatchedProjects/master</url>
+        </pluginRepository>
+    </pluginRepositories>
+----
+
+The Notifier API itself is available in the `internal-api` artifact that is located in our Public Nexus repository.
+
+----
+   <dependency>
+       <groupId>fish.payara.server.internal.common</groupId>
+       <artifactId>internal-api</artifactId>
+       <version>5.2020.5</version>
+       <optional>true</optional>
+   </dependency>
+----
+
+The Notifier API did not change since it was released, the above dependency can be used to create a notifier that can be used on all Community versions since 5.2020.5.
```

Comment:
Does this take into account the issues with the latest release? I believe you have to be using the same minor version as the release that you intend to deploy to

Reply:
I did not test with 5.2021.2 but it worked just fine on 5.2021.1

And since only 5.2020.5 is available on public Nexus, people doesn't have any choice and can't match the version.